userprofile/models.py

```diff
@@ -84,7 +85,11 @@ def is_external(self):
         """
         Is this an external rather than internal account.
         """
-        return not bool(self.student_id)
+        # if organization is not set (old accounts), apply legacy logic based on student_id
+        if not self.organization:
```

Comment:
We may have some (teacher's) features that call `is_external`. If the student has never logged in after the server upgrade, your implementation would classify many local students as external if the students do not log in at all in the summer. (We are going to upgrade the servers during the summer break when many students are away.)

Reply:
Now I made a mistake in the previous comment. If we have the if branches in `ìs_external`, then (Aalto) users with empty organization are still labelled "Aalto" since they have the student id. 

Though, if we have local accounts (with local passwords), then those do not have any organization, but some of those accounts may have a fake student id in the database.